File: TypeScript/google_search_cli.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var child_process_1 = require("child_process");
var readline = require("readline");
var rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});
rl.question('Enter your search query: ', function (query) {
    var url = "https://www.google.com/search?q=".concat(encodeURIComponent(query));
    var command;
    if (process.platform === 'win32') {
        command = "start \"\" \"".concat(url, "\"");
    }
    else if (process.platform === 'darwin') {
        command = "open \"".concat(url, "\"");
    }
    else {
        command = "xdg-open \"".concat(url, "\"");
    }
    (0, child_process_1.exec)(command, function (err) {
        if (err) {
            console.error('Failed to open browser:', err);
        }
        else {
            console.log("Searching for: ".concat(query));
        }
    });
    rl.close();
});

File: GoogleSearchCLI.scala
import java.awt.Desktop
import java.net.URI
import scala.io.StdIn.readLine

object GoogleSearchCLI {
  def main(args: Array[String]): Unit = {
    print("Enter your search query: ")
    val query = readLine()

    val url = s"https://www.google.com/search?q=${query.replace(" ", "+")}"

    try {
      val desktop = Desktop.getDesktop
      desktop.browse(new URI(url))
      println(s"Searching for: $query")
    } catch {
      case e: Exception => e.printStackTrace()
    }
  }
}
